base/platform/linux/base_cellular_utilities_linux.cpp

```diff
@@ -0,0 +1,159 @@
+// This file is part of Desktop App Toolkit,
+// a set of libraries for developing nice desktop applications.
+//
+// For license and copyright information please follow this link:
+// https://github.com/desktop-app/legal/blob/master/LEGAL
+//
+#include "base/platform/linux/base_cellular_utilities_linux.h"
+
+#include "base/const_string.h"
+
+#ifndef DESKTOP_APP_DISABLE_DBUS_INTEGRATION
+#include "base/platform/linux/base_linux_glibmm_helper.h"
+
+#include <glibmm.h>
+#include <giomm.h>
+#endif // !DESKTOP_APP_DISABLE_DBUS_INTEGRATION
+
+namespace base::Platform {
+namespace {
+
+constexpr auto kMMService = "org.freedesktop.ModemManager1"_cs;
+
+typedef enum {
+	MM_MODEM_MODE_NONE = 0,
+	MM_MODEM_MODE_CS   = 1 << 0,
+	MM_MODEM_MODE_2G   = 1 << 1,
+	MM_MODEM_MODE_3G   = 1 << 2,
+	MM_MODEM_MODE_4G   = 1 << 3,
+	MM_MODEM_MODE_5G   = 1 << 4,
+	MM_MODEM_MODE_ANY  = 0xFFFFFFFF
+} MMModemMode;
```

Comment:
Why not `enum class`?

Reply:
It's copied from https://gitlab.freedesktop.org/mobile-broadband/ModemManager/-/blob/master/include/ModemManager-enums.h to avoid dependency on libmm-glib just for that include

Reply:
So perhaps it should be commented.

Reply:
How will it work in commented state?

Reply:
@ilya-fedin I daresay @23rd suggested to document in a comment where you took these enums from. At least you must mention original authors because their code is under GPLv2+.

Reply:
> I daresay @23rd suggested to document in a comment where you took these enums from.

Ah

> At least you must mention original authors because their code is under GPLv2+.

I don't think you really can license enum declaration (see e.g. Google vs Oracle)